terraform-modules/external-dns/variables.tf

```diff
@@ -0,0 +1,10 @@
+# google project
+variable "target_project" {}
+#region
+variable "region" {}
+#credentials
+variable "target_credentials" {}
+#name of external dns_zone name
+variable "target_dns_zone_name" {}
+#name and destination values ie: record = 192.168.44.22
+variable "records" {}
```

Comment:
I think you can give a type annotation here to help people lock down the valid key names, like:
```
type = map(object({type = string, rrdatas = string}))
```